pandas/tests/indexes/datetimes/test_partial_slicing.py

```diff
@@ -349,7 +349,7 @@ def test_loc_datetime_length_one(self):
 
     @pytest.mark.parametrize('datetimelike', [
         Timestamp('20130101'), datetime(2013, 1, 1),
-        date(2013, 1, 1), np.datetime64('2013-01-01T00:00', 'ns')])
+        np.datetime64('2013-01-01T00:00', 'ns')])
```

Comment:
Why did you have to change the parametrization?

Reply:
b/c ATM `Series[datetime64].__cmp__(date)` treats the `date` as a `datetime`, i.e. allows the comparison.  But `DatetimeIndex` does not -- following convention set by `Timestamp` (and `datetime` itself).  The `DatetimeIndex` behavior is canonical.